src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -244,7 +249,10 @@ object ProtocPlugin extends AutoPlugin with Compat {
       log.info("Compiling protobuf")
       sortedTargetPaths.foreach { path => log.info("Protoc target: %s".format(path.absolutePath)) }
 
-      targets.flatMap { ot => (ot.outputPath ** ("*.java" | "*.scala")).get }.toSet
+      targets.flatMap {
+        case Target(DescriptorSetGenerator(), outputFile, _) => Seq(outputFile)
+        case Target(_, outputDirectory, _)                   => outputDirectory.allPaths.get
```

Comment:
This may capture files that protoc didn't generate in case `deleteTargetDirectory` is set to false, but this would only have an impact on clients directly using the return value of `protocGenerate`, so I think it's acceptable. I contemplated doing a filesystem diff on the target directories before/after the `protoc` run to filter out what was there already, but it seems unnecessarily complex.